libexec/pyenv-init

```diff
@@ -260,52 +253,40 @@ function print_shell_function() {
   commands=(`pyenv-commands --sh`)
   case "$shell" in
   fish )
-    cat <<EOS
-function pyenv
-  set command \$argv[1]
+    echo 'function pyenv
+  set command $argv[1]
   set -e argv[1]
 
-  switch "\$command"
-  case ${commands[*]}
-    source (pyenv "sh-\$command" \$argv|psub)
-  case '*'
-    command pyenv "\$command" \$argv
+  switch "$command"
+  case '"${commands[*]}"'
+    source (pyenv "sh-$command" $argv|psub)
+  case "*"
+    command pyenv "$command" $argv
   end
-end
-EOS
+end'
     ;;
   ksh | ksh93 | mksh )
-    cat <<EOS
-function pyenv {
-  typeset command
-EOS
+    echo 'function pyenv {
+  typeset command=$1'
```

Comment:
Need to retain `$(1:-}` because the resulting shell function runs in user's shell which may have `set -u` active. Then this'll fail with `$1: unbound variable`